be/src/runtime/mem_pool.h

```diff
@@ -243,6 +243,37 @@ class MemPool {
     std::vector<ChunkInfo> chunks_;
 };
 
+// Custom allocator based on mempool
+template <class T, typename Base = std::allocator<T>>
+class pool_allocator : public Base {
+public:
+    typedef std::allocator_traits<Base> b_t;
+
+    pool_allocator(MemPool* mem) : _mem(mem) {}
+    ~pool_allocator() = default;
+
+    using Base::Base;
+
+    template <typename U>
+    struct rebind {
+        using other = pool_allocator<U, typename b_t::template rebind_alloc<U>>;
+    };
+
+    // allocate more than caller required
+    T* allocate(size_t n) { return (T*)_mem->allocate(n); }
```

Comment:
slice.data = reinterpret_cast<char*>(mem_pool->allocate(slice.size));
            if (UNLIKELY(slice.data == nullptr)) {
                return Status::MemoryLimitExceeded("Mem usage has exceed the limit of BE");
            }

@trueeyu do we need check memory limit here？

Reply:
should check

Reply:
Good remind.

To deal with mempool allocation failure:
1. throw bad_alloc in allocator if allocate failed
2. check allocated pointer in `JsonValue::clone`